packages/tools/src/tools/SculptorTool.ts

```diff
@@ -0,0 +1,590 @@
+      toolIndex: undefined,
+      annotationUID: undefined,
+    };
+
+    for (let i = 0; i < annotations?.length; i++) {
+      if (annotations[i].isLocked || !annotations[i].isVisible) {
+        continue;
+      }
+
+      const distanceFromTool = distanceFromPoint(
+        viewport,
+        annotations[i].data,
+        canvasPoints
+      );
+
+      if (distanceFromTool === -1) {
+        continue;
+      }
+
+      if (distanceFromTool < closest.distance) {
+        closest.distance = distanceFromTool;
+        closest.toolIndex = i;
+        closest.annotationUID = annotations[i].annotationUID;
+      }
+    }
+
+    this.commonData.isEditingOpenContour =
+      !annotations[closest.toolIndex].data.contour.closed;
+
+    config.referencedToolName =
+      annotations[closest.toolIndex].metadata.toolName;
+
+    return closest.annotationUID;
+  }
+
+  /**
+   * Event handler for MOUSE_UP during the active loop.
+   *
+   * @param evt - The event
+   */
+  private endCallback = (
+    evt: EventTypes.MouseUpEventType | EventTypes.MouseClickEventType
+  ): void => {
+    const eventData = evt.detail;
+    const { element } = eventData;
+    const config = this.configuration;
+    const enabledElement = getEnabledElement(element);
+
+    this.isActive = false;
+    this.deactivateSculpt(element);
+    resetElementCursor(element);
+
+    const { renderingEngineId, viewportId } = enabledElement;
+
+    const toolGroup = ToolGroupManager.getToolGroupForViewport(
+      viewportId,
+      renderingEngineId
+    );
+
+    const toolInstance = toolGroup.getToolInstance(config.referencedToolName);
+
+    const annotations = this.filterSculptableAnnotationsForElement(element);
+
+    const activeAnnotation = annotations.find(
+      (annotation) =>
+        annotation.annotationUID === this.commonData.activeAnnotationUID
+    );
+
+    if (toolInstance.configuration.calculateStats) {
+      activeAnnotation.invalidated = true;
+    }
+
+    triggerAnnotationModified(activeAnnotation, element);
+  };
+
+  /**
+   * Event handler for MOUSE_DRAG during the active loop.
+   *
+   * @param evt - The event
+   */
+  private dragCallback(evt: EventTypes.InteractionEventType): void {
+    const eventData = evt.detail;
+    const element = eventData.element;
+
+    this.updateCursor(evt);
+
+    const annotations = this.filterSculptableAnnotationsForElement(element);
+    const activeAnnotation = annotations.find(
+      (annotation) =>
+        annotation.annotationUID === this.commonData.activeAnnotationUID
+    );
+
+    if (!annotations?.length || !this.isActive) {
+      return;
+    }
+
+    const points = activeAnnotation.data.contour.polyline;
+
+    this.sculpt(eventData, points);
+  }
+
+  /**
+   * Attaches event listeners to the element such that is is visible, modifiable, and new data can be created.
+   * @param element - - The viewport element to attach event listeners to.
+   */
+  private activateSculpt(element: HTMLDivElement): void {
```

Comment:
Can we use the generic names on this - which are:
_activateModify
_deactivateModify
See AdvancedMagnifyTool for one example.

Also, the methods should be field methods -that causes the bindings to be held, so that when you add it as a listener, you don't need to rebind it, that is:
private _activateModify = (element: HTMLDivElement) => {
....};